moped-editor/src/views/projects/projectView/ProjectComponents/TheMap.js

```diff
@@ -191,8 +180,83 @@ export default function TheMap({
       ctnLinesGeojson
     );
 
-    newDraftComponent.features.push(newFeature);
-    setDraftComponent(newDraftComponent);
+    const tableToInsert =
+      draftEditComponent?.moped_components?.feature_layer?.internal_table;
+
+    // Update UI
+    const addOrRemoveClickedEditFeatures = (currentComponent) => {
+      const isFeatureAlreadyInComponent = Boolean(
+        currentComponent[tableToInsert].find(
+          (feature) =>
+            feature?.[sourceFeatureId.toLowerCase()] === featureUniqueId || // Already in database
+            feature?.properties?.[sourceFeatureId] === featureUniqueId // From CTN layers
+        )
+      );
+
+      // If the feature is not already in the draftEditComponent, add it
+      // otherwise, remove it.
+      if (!isFeatureAlreadyInComponent) {
+        return {
+          ...currentComponent,
+          [tableToInsert]: [...currentComponent[tableToInsert], newFeature],
+        };
+      } else if (isFeatureAlreadyInComponent) {
+        const filteredFeatures = currentComponent[tableToInsert].filter(
+          (feature) =>
+            feature?.[sourceFeatureId.toLowerCase()] !== featureUniqueId && // Already in database
+            feature?.properties?.[sourceFeatureId] !== featureUniqueId // From CTN layers
+        );
+
+        const isAtLeastOneFeatureRemaining = filteredFeatures.length > 0;
```

Comment:
Make sure that at least one feature is left when editing. Maybe this could show up in the UI down the road as a warning modal or something. For now, the UI just won't unselect a feature if it is the last one standing.

Reply:
I don't think that would be a large lift at all. Backlog sounds good for this one too. I caught this late in the game so patched it real quick. Disabling the button just like on creation definitely feels right.

Reply:
👍 https://github.com/cityofaustin/atd-data-tech/issues/10991